dashboard/lib/announcements.rb

```diff
@@ -0,0 +1,60 @@
+class Announcements
+  @@announcements_data = nil
+  @@loaded = false
+  @@load_error = false
+  @@json_path = File.join("#{pegasus_dir}/sites.v3/code.org", 'announcements.json')
```

Comment:
This is redundant - if you walk down into the implementation of `pegasus_dir` it is, itself, a currying of `File.join`.  So you can write this instead:
```suggestion
  @@json_path = pegasus_dir 'sites.v3', 'code.org', 'announcements.json'
```

Or even
```rb
  @@json_path = pegasus_dir 'sites.v3/code.org/announcements.json'
```